types/astraled/mantis/uplink.js

```diff
@@ -0,0 +1,229 @@
+// Decoder Gateway Protocol Version 0.02
+function swap16(val) {
+  return ((val & 0xFF) << 8) |
+    ((val >> 8) & 0xFF);
+}
+
+function swap32(val) {
+  return (((val << 24) & 0xff000000) |
+    ((val << 8) & 0x00ff0000) |
+    ((val >> 8) & 0x0000ff00) |
+    ((val >> 24) & 0x000000ff));
+}
+
+function bytesToFloat(bytes) {
+  var bits_float = bytes;
+  var sign = (bits_float >>> 31 === 0) ? 1.0 : -1.0;
+  var e = bits_float >>> 23 & 0xff;
+  var m = (e === 0) ? (bits_float & 0x7fffff) << 1 : (bits_float & 0x7fffff) | 0x800000;
+  var f = sign * m * Math.pow(2, e - 150);
+  return f;
+}
+
+function getByte(buff, nr) {
+  return Bits.bitsToUnsigned(buff.substr(nr * 8, 8));
+}
+
+function get_word(buff, nr) {
+  return swap16(Bits.bitsToUnsigned(buff.substr(nr * 8, 16)));
+}
+
+function get_long(buff, nr) {
+  return swap32(Bits.bitsToUnsigned(buff.substr(nr * 8, 32)));
+}
+
+function get_float(buff, nr) {
```

Comment:
🐫

Reply:
please use a linter to make sure these things pop up during writing code, call me if you need help